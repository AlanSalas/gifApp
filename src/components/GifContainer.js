import { useState, useEffect } from "react";
import Gif from "./Gif";

const GifContainer = ({ category }) => {
  const [gifs, setGifs] = useState([]);

  const fetchData = async () => {
    const res = await fetch(
      `https://api.giphy.com/v1/gifs/search?q=${encodeURI(
        category
      )}&limit=10&api_key=7IGuswnWXqS56OHmYZc4hbHi4XsdFiaD`
    );

    const { data } = await res.json();

    const gifsData = data.map((gif) => {
      return {
        id: gif.id,
        title: gif.title,
        url: gif.images.original.url,
      };
    });

    console.log(gifsData);
    setGifs(gifsData);
  };

  useEffect(() => {
    fetchData();
  }, []);

  return (
    <>
      <h3>{category}</h3>
      <div className="gif-container">
        {gifs.map((gif) => (
          <Gif key={gif.id} {...gif} />
        ))}
      </div>
    </>
  );
};

export default GifContainer;
