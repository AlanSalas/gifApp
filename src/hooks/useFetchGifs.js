import { useState, useEffect } from "react";
import { getGifs } from "../api/gifs";

const useFetchGifs = (category) => {
  const [data, setData] = useState({
    gifs: [],
    loading: true,
  });

  useEffect(() => {
    getGifs(category).then((gifs) => {
      setData({
        gifs,
        loading: false,
      });
    });
  }, [category]);

  return data;
};

export default useFetchGifs;
