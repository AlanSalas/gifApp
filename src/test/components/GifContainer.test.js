import { shallow } from "enzyme";
import GifContainer from "../../components/GifContainer";
import useFetchGifs from "../../hooks/useFetchGifs";
jest.mock("../../hooks/useFetchGifs");

describe("Tests in GifContainer Component", () => {
  test("Should render properly", () => {
    useFetchGifs.mockReturnValue({
      gifs: [],
      loading: true,
    });
    const wrapper = shallow(<GifContainer category="Messi" />);
    expect(wrapper).toMatchSnapshot();
  });

  test("Should render items with the return data of useFetchGifs", () => {
    const gifs = [
      {
        id: "test",
        title: "test",
        url: "https://test.com/image/test.gif",
      },
      {
        id: "123",
        title: "test",
        url: "https://test.com/image/test.gif",
      },
    ];
    useFetchGifs.mockReturnValue({
      gifs,
      loading: false,
    });
    const wrapper = shallow(<GifContainer category="Messi" />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("Spinner").exists()).toBe(false);
    expect(wrapper.find("Gif").length).toBe(gifs.length);
  });
});
