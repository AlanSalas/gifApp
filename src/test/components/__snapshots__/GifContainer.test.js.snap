// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in GifContainer Component Should render items with the return data of useFetchGifs 1`] = `
<div
  className="gif-container"
>
  <h3
    className="gif-container__title"
  >
    Results for 
    Messi
  </h3>
  <Masonry
    breakpointCols={
      Object {
        "1100": 3,
        "500": 1,
        "700": 2,
        "default": 3,
      }
    }
    className="my-masonry-grid"
    columnClassName="my-masonry-grid_column"
  >
    <Gif
      id="test"
      key="test"
      title="test"
      url="https://test.com/image/test.gif"
    />
    <Gif
      id="123"
      key="123"
      title="test"
      url="https://test.com/image/test.gif"
    />
  </Masonry>
</div>
`;

exports[`Tests in GifContainer Component Should render properly 1`] = `
<div
  className="gif-container"
>
  <h3
    className="gif-container__title"
  >
    Results for 
    Messi
  </h3>
  <Spinner />
  <Masonry
    breakpointCols={
      Object {
        "1100": 3,
        "500": 1,
        "700": 2,
        "default": 3,
      }
    }
    className="my-masonry-grid"
    columnClassName="my-masonry-grid_column"
  />
</div>
`;
